src/main/org/pantsbuild/jarjar/MainProcessor.java

```diff
@@ -52,7 +54,7 @@ public MainProcessor(List<PatternElement> patterns, boolean verbose, boolean ski
         if (kp != null)
             processors.add(kp);
         processors.add(new ZapProcessor(zapList));
-        processors.add(new JarTransformerChain(new RemappingClassTransformer[]{ new RemappingClassTransformer(pr) }));
```

Comment:
I could give `EntryStruct` a `shouldSkip` boolean that the separate process could check, but that feels hacky. Alternatively a `MisplacedClassProcessor` step could populate a `Map` somewhere, but that just replaces the `MisplacedClassStrategy` plumbing with plumbing for the `Map`.

I could wrap the `strategy` in a generic `ProcessingContext` object which would have the same purpose like the `Context` object used in pants, which would be effectively the same thing I have now but would perhaps look less ad-hoc.